### PR TITLE
Add eslint-config-preact and fix found issues

### PR DESCRIPTION
**Pull request by sjinks**
## Description

This PR adds `eslint-config-preact` and fixes the issues found by the linter (mainly `react-hooks/exhaustive-deps`). Also, this PR removes all uses of `dangerouslySetInnerHTML`.

### Plugin Updated: Search Dev Tools

Minor bug fixes.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

E2E tests should pass.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   "author": "Automattic",
   "devDependencies": {
     "@automattic/eslint-plugin-wpvip": "^0.3.0",
+    "@babel/plugin-syntax-decorators": "^7.18.6",
```
Comment (sjinks) [inline package.json:21]: Unfortunately, this is required for the `eslint` VS Code extension to work: it appears to load Babel plugins from `node_modules` relative to the project root directory, not relative to `.eslintrc`.

File: search/search-dev-tools/src/components/query/index.js [hunk 2]
```diff
@@ -42,28 +42,27 @@ const Query = ( { args, request, url, query_args, backtrace = [] } ) => {
...
+	const fetchForQuery = useCallback( async query => {
 		try {
 			const res = await postData( window.VIPSearchDevTools.ajaxurl, {
 				action: window.VIPSearchDevTools.action,
 				url,
 				query,
 			}, window.VIPSearchDevTools.nonce );
 
-			setState( { ...state, result: JSON.stringify( res?.result?.body, null, 2 ) } );
+			setState( prevState => ( { ...prevState, result: JSON.stringify( res?.result?.body, null, 2 ) } ) );
```
Comment (sjinks) [inline search/search-dev-tools/src/components/query/index.js:55]: This eliminates the dependency on `state`.

File: search/search-dev-tools/src/components/query/index.js [hunk 3]
```diff
@@ -159,10 +156,10 @@ const Query = ( { args, request, url, query_args, backtrace = [] } ) => {
 			<div className={ style.query_res }>
 				<div className={ style.query_result }>
 					<div className={ style.query_actions }>
-						<button id="query-response-copy-handle" data-clipboard-target="#query-response-text" dangerouslySetInnerHTML={ { __html: 'COPY' } }></button>
+						<button id="query-response-copy-handle" data-clipboard-target="#query-response-text">COPY</button>
 					</div>
 					<pre className="line-numbers">
-						<code className="language-json" ref={ queryResultRef } id="query-response-text" dangerouslySetInnerHTML={ { __html: state.result } }></code>
+						<code className="language-json" id="query-response-text">{ state.result }</code>
```
Comment (sjinks) [inline search/search-dev-tools/src/components/query/index.js:162]: 1. We don't use the `ref` anywhere, let's drop it.
2. `state.result` is a plain text (the result of `JSON.stringify()`), there is no need to use `dangerouslySetInnerHTML`.

File: search/search-dev-tools/src/components/query/index.js [hunk 4]
```diff
@@ -159,10 +156,10 @@ const Query = ( { args, request, url, query_args, backtrace = [] } ) => {
 			<div className={ style.query_res }>
 				<div className={ style.query_result }>
 					<div className={ style.query_actions }>
-						<button id="query-response-copy-handle" data-clipboard-target="#query-response-text" dangerouslySetInnerHTML={ { __html: 'COPY' } }></button>
+						<button id="query-response-copy-handle" data-clipboard-target="#query-response-text">COPY</button>
```
Comment (sjinks) [inline search/search-dev-tools/src/components/query/index.js:159]: There is no need to use `dangerouslySetInnerHTML` for a text string.

**Review by rebeccahum — APPROVED** [review]

**Pull request merged**